app/parse.py

```diff
@@ -1,15 +1,59 @@
+import csv
+import requests
+from bs4 import BeautifulSoup
 from dataclasses import dataclass
+from typing import List
+
+
+BASE_URL = "https://quotes.toscrape.com"
 
 
 @dataclass
 class Quote:
     text: str
     author: str
-    tags: list[str]
+    tags: List[str]
+
+
+def parse_single_quote(quote_element) -> Quote:
+    text = quote_element.select_one(".text").get_text()
+    author = quote_element.select_one(".author").get_text()
+    tags = [tag.get_text() for tag in quote_element.select("a.tag")]
+    return Quote(text=text, author=author, tags=tags)
+
+
+def parse_quotes_from_page(page_url: str) -> List[Quote]:
+    response = requests.get(page_url)
+    response.raise_for_status()
+    soup = BeautifulSoup(response.text, "html.parser")
+    quote_elements = soup.find_all("div", class_="quote")
+    return [parse_single_quote(quote_element) for quote_element in quote_elements]
+
+
+def get_all_quotes() -> List[Quote]:
+    all_quotes = []
+    page_url = "/page/1"
+
+    while page_url:
+        quotes = parse_quotes_from_page(BASE_URL + page_url)
+        all_quotes.extend(quotes)
+
+        response = requests.get(BASE_URL + page_url)
```

Comment:
This request to fetch the page content is redundant because the same request is already made in the `parse_quotes_from_page` function. You can reuse the `response` object from that function to avoid making an unnecessary HTTP request.